keps/sig-node/20190717-seccomp-ga.md

```diff
@@ -387,9 +481,228 @@ alpha](#alternatives).
+  
+  This feature was implemented prior to the concept of feature gates were adopted 
+  within Kubernetes. This KEP is simply "cleaning up" the API to make it GA.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we rollback
+  the enablement)?**
+  Also set `rollback-supported` to `true` or `false` in `kep.yaml`.
+  Describe the consequences on existing workloads (e.g. if this is runtime
+  feature, can it break the existing applications?).
+
+  This feature has no `rollback support`. The impact on existing workloads is defined 
+  under [Version Skew Strategy](#version-skew-strategy).
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+  
+  N/A - the feature is already enabled by default since Kubernetes 1.3.
+
+* **Are there any tests for feature enablement/disablement?**
+  The e2e framework does not currently support enabling and disabling feature
+  gates. However, unit tests in each component dealing with managing data created
+  with and without the feature are necessary. At the very least, think about
+  conversion tests if API types are being modified.
+
+  N/A - the feature is already enabled by default since Kubernetes 1.3.
+
+### Rollout, Upgrade and Rollback Planning
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  Try to be as paranoid as possible - e.g. what if some components will restart
+  in the middle of rollout?
+
+  The [Version Skew Strategy](#version-skew-strategy) section covers this point.
+  Running workloads should have no impact as the Kubelet will support either the 
+  existing annotations or the new fields introduced by this KEP. 
+
+* **What specific metrics should inform a rollback?**
+
+  N/A - the feature is already enabled by default since Kubernetes 1.3.
+
+* **Were upgrade and rollback tested? Was upgrade->downgrade->upgrade path tested?**
+  Describe manual testing that was done and the outcomes.
+  Longer term, we may want to require automated upgrade/rollback tests, but we
+  are missing a bunch of machinery and tooling and do that now.
+
+  Automated tests will cover the scenarios with and without the changes proposed 
+  on this KEP. As defined under [Version Skew Strategy](#version-skew-strategy),
+  we are assuming the cluster may have kubelets with older versions (without 
+  this KEP' changes), therefore this will be covered as part of the new tests.
+
+### Monitoring requirements
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can an operator determine if the feature is in use by workloads?**
+  Ideally, this should be a metrics. Operations against Kubernetes API (e.g.
+  checking if there are objects with field X set) may be last resort. Avoid
+  logs or events for this purpose.
+
+  The feature is built into the kubelet and api server components. No metric is
```

Comment:
@tallclair do you think it may be useful to have metrics around seccomp usage (i.e. # pods per profile type)? This could potentially help cluster admins to understand seccomp gap/cover. 

But feels like something that could be explored on seccomp-operator instead.

Reply:
Agreed. Metrics for pod feature usage or usually recorded at the control-plane level rather than by the kubelet.